src/youtube-player/README.md

```diff
@@ -0,0 +1,124 @@
+# Angular YouTube Player component
+
+This component provides a simple angular wrapper around the embed [YouTube player API](https://developers.google.com/youtube/iframe_api_reference). File any bugs against the [angular/components repo](https://github.com/angular/components/issues).
+
+## Installation
+
+To install, run `npm install @angular/youtube-player`.
+
+## Usage
+
+Follow the following instructions for setting up the YouTube player component:
+
+- First, follow the [instructions for installing the API script](https://developers.google.com/youtube/iframe_api_reference#Getting_Started).
+- Then make sure the API is available before bootstraping the YouTube Player component.
+- Provide find the video id by extracting it from the video URL.
+
+## Example
+
+If you're video is found at https://www.youtube.com/watch?v=PRQCAL_RMVo, then your video id is `PRQCAL_RMVo`.
+
+```typescript
+// example-module.ts
+import {CommonModule} from '@angular/common';
+import {NgModule} from '@angular/core';
+import {YouTubePlayerModule} from '@angular/youtube-player';
+import {YoutubePlayerExample} from './example-component';
+
+@NgModule({
+  imports: [
+    // Needed for ngIf directive.
+    CommonModule,
+    YouTubePlayerModule,
+  ],
+  declarations: [YoutubePlayerExample],
+})
+export class YoutubePlayerExampleModule {
+}
+
+// example-component.ts
+@Component({
+  moduleId: module.id,
+  template: '<youtube-player videoId="PRQCAL_RMVo" *ngIf="apiReady" />',
+  selector: 'youtube-player-example',
+})
+class YoutubePlayerExample implements OnInit {
+  /** Field preventing the YouTube player from bootstrapping until the api is ready. */
+  apiReady = false;
+
+  constructor(private _ref: ChangeDetectorRef) {}
+
+  onInit() {
+    // This code loads the IFrame Player API code asynchronously, according to the instructions at
+    // https://developers.google.com/youtube/iframe_api_reference#Getting_Started
+    var tag = document.createElement('script');
```

Comment:
Use `const` instead of `var`.

Reply:
Also if somebody took this example verbatim, wouldn't they get a new `script` tag for every time the user visits a page? Couldn't we somehow handle this for the consumer from inside the YouTube player package?

Reply:
> Also if somebody took this example verbatim, wouldn't they get a new script tag for every time the user visits a page?

They would get a new script tag for every instance of this example component that they inject.

> Couldn't we somehow handle this for the consumer from inside the YouTube player package?

The problem is that this is not necessarily the technique that the user would use to inject the script into their page. They could bundle the script into their binary, or download it via npm.

If we try to resolve this problem on their behalf, we then have to figure out whether or not the script has already been loaded, which is impossible if it's in the process of loading and hasn't loaded yet. This would require some kind of inter-component coordination, probably through a service. Anyways, it's certainly an option.